### ISSUE TITLE
Some default gems aren't activated

### ISSUE THREAD

**Issue by andrykonchin** [oracle/truffleruby#3663]
Some default gems (some of gems with native extensions and with custom TruffleRuby-specific implementation) aren't activated (RubyGems' term) when are required.

It could lead, for instance, to excessive warnings (example with `bundler-audit` gem that depends on non-default `stringio` version):

```
$ bundle exec rake test
WARN: Unresolved or ambiguous specs during Gem::Specification.reset:
      stringio (>= 0)
      Available/installed versions of this gem:
      - 3.1.1
      - 3.0.4
```

The issue affects the following default gems:
- etc
- fcntl
- io-console
- io-nonblock
- io-wait
- nkf
- stringio
- strscan
- syslog
- zlib

**Comment by andrykonchin** [comment]
Fixed in 92c6d2ec3e94d3607ddf5d7b1c280021b9faca89

**Comment by rosingrind** [comment]
Is there a way to disable this warning in CLI of `truffleruby-24.1.1` when performing install/update actions?

**Comment by eregon** [comment]
I'm not aware of any, no, sorry.
You can use truffleruby-head to get the fix.
